uberfire-structure/uberfire-structure-backend/src/main/java/org/guvnor/structure/backend/organizationalunit/OrganizationalUnitFactoryImpl.java

```diff
@@ -37,88 +34,44 @@
-        this.configurationService = configurationService;
-        this.configurationFactory = configurationFactory;
     }
 
     @Override
-    public OrganizationalUnit newOrganizationalUnit(ConfigGroup groupConfig) {
+    public OrganizationalUnit newOrganizationalUnit(final SpaceInfo spaceInfo) {
 
-        OrganizationalUnitImpl organizationalUnit = new OrganizationalUnitImpl(groupConfig.getName(),
-                                                                               groupConfig.getConfigItemValue("defaultGroupId"));
+        OrganizationalUnitImpl organizationalUnit = new OrganizationalUnitImpl(spaceInfo.getName(),
+                                                                               spaceInfo.getDefaultGroupId(),
+                                                                               spaceInfo.isDeleted());
 
-        ConfigItem<List<String>> repositories = groupConfig.getConfigItem("repositories");
+        final List<RepositoryInfo> repositories = spaceInfo.getRepositories();
         if (repositories != null) {
-            for (String alias : repositories.getValue()) {
+            for (RepositoryInfo alias : repositories) {
                 Space space = spacesAPI.getSpace(organizationalUnit.getName());
                 final Repository repo = repositoryService.getRepositoryFromSpace(space,
-                                                                                 alias);
+                                                                                 alias.getName());
                 if (repo != null) {
                     organizationalUnit.getRepositories().add(repo);
                 }
             }
         }
 
-        //Copy in Security Roles required to access this resource
-        ConfigItem<List<String>> groups = backward.compat(groupConfig).getConfigItem("security:groups");
-        if (groups != null) {
-            for (String group : groups.getValue()) {
-                organizationalUnit.getGroups().add(group);
-            }
+        final List<String> securityGroups = spaceInfo.getSecurityGroups();
+        if (securityGroups != null) {
+            organizationalUnit.getGroups().addAll(securityGroups);
         }
 
-        fillOrganizationalUnitContributors(groupConfig, organizationalUnit);
-
-        return organizationalUnit;
-    }
-
-    private void fillOrganizationalUnitContributors(final ConfigGroup configGroup,
-                                                    final OrganizationalUnit organizationalUnit) {
-        boolean shouldUpdateConfigGroup = false;
-
-        final String oldOwner = configGroup.getConfigItemValue("owner");
-        if (oldOwner != null) {
-            shouldUpdateConfigGroup = true;
-            organizationalUnit.getContributors().add(new Contributor(oldOwner, ContributorType.OWNER));
-            configGroup.removeConfigItem("owner");
+        final Collection<Contributor> contributors = spaceInfo.getContributors();
```

Comment:
all these logic was moved to getContributors?

Reply:
Actually this was something old. Now is replaced during migration.Also, after removing this from here, I've never seen errors in smoke tests